File: onemoretime/polynomial/polynomial.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class polynomial {

    public static double[] interpolate(int d, List<Double> x, List<Double> y) {
        double[] p = new double[d];

        for (int i = 0; i < d; i++) {
            double product = 1.0;
            double[] t = new double[d];

            for (int j = 0; j < d; j++) {
                if (i != j) {
                    product *= (x.get(i) - x.get(j));
                }
            }

            product = y.get(i) / product;
            t[0] = product;

            for (int j = 0; j < d; j++) {
                if (i != j) {
                    for (int k = d - 1; k > 0; k--) {
                        t[k] += t[k - 1];
                        t[k - 1] *= (-x.get(j));
                    }
                }
            }

            for (int j = 0; j < d; j++) {
                p[j] += t[j];
            }
        }

        return p;
    }

    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Error: Please provide a file path.");
            return;
        }

        String path = args[0];
        File file = new File(path);
        if (!file.exists()) {
            System.out.printf("Error: The file at '%s' doesn't exist!\n", path);
            return;
        }

        List<Double> x = new ArrayList<>();
        List<Double> y = new ArrayList<>();

        try (Scanner scanner = new Scanner(file)) {
            while (scanner.hasNextLine()) {
                String line = scanner.nextLine();
                String[] split = line.split(" ");

                if (split.length != 2) {
                    System.out.println("Error: Invalid input format");
                    return;
                }

                x.add(Double.parseDouble(split[0]));
                y.add(Double.parseDouble(split[1]));
            }

            double[] p = interpolate(x.size(), x, y);
            for (double val : p) {
                System.out.print(val + " ");
            }
            System.out.println();
        } catch (FileNotFoundException e) {
            System.out.printf("Error: The file at '%s' doesn't exist!\n", path);
        } catch (NumberFormatException e) {
            System.out.println("Error: Invalid number format in the file");
        }
    }
}

File: 08. One More Time/polynomial/Python 3.X/polynomial.py
import argparse
import os

def interpolate(d : int, x : list[float], y : list[float]):
    
    p = [0 for i in range(d)]
    
    for i in range(d):
        product = 1.0
        for j in range(d):
            t = [0 for i in range(d)]
        for j in range(d):
            if (i == j):
                continue
            product *= (x[i] - x[j])
        product = y[i] / product
        t[0] = product
        for j in range(d):
            if (i == j):
                continue
            for k in range(d - 1,0,-1):
                t[k] += t[k - 1]
                t[k - 1] *= (-x[j])
        for j in range(d):
            p[j] += t[j]
    
    return p

def main():
    # Initialize the parser
    d = 0
    x = []
    y = []

    parser = argparse.ArgumentParser(description="Process the path argument.")
    
    parser.add_argument("path", type=str, help="Path to the file")

    args = parser.parse_args()

    if args.path:
        if not os.path.exists(args.path):
            print(f"Error: The file at '{args.path}' doesn't exist!")
            return
        else:
            f = open(args.path, "r")
            try:
                for line in f:
                    split = line.split(" ")
                    if len(split) != 2:
                        raise ValueError("")
                    x.append(float(split[0]))
                    y.append(float(split[1]))
                    d += 1
                p = interpolate(d,x,y)
                for i in p:
                    print(i,end=" ")
                print()
            except:
                print("Error: Invalid input format")

if __name__ == "__main__":
    main()